src/domain/individualHarbor/forms/berthForm/BerthForm.tsx

```diff
@@ -0,0 +1,210 @@
+            },
+          },
+        }).then(() => onUpdate && onUpdate(values));
+      }}
+    >
+      {({ values, errors, handleChange, handleSubmit, setFieldValue }) => (
+        <form onSubmit={handleSubmit} className={styles.form}>
+          <Grid colsCount={3} className={styles.grid}>
+            <TextInput
+              id="pier"
+              readOnly
+              value={values.pier}
+              labelText={t('forms.berth.pier')}
+              invalid={!!errors.pier}
+              invalidText={errors.pier}
+            />
+            <TextInput
+              id="number"
+              type="number"
+              readOnly
+              value={String(values.number)}
+              labelText={t('forms.berth.number')}
+              invalid={!!errors.number}
+              invalidText={errors.number}
+            />
+            <div />
+
+            <TextInput
+              id="width"
+              type="number"
+              onChange={handleChange}
+              value={String(values.width)}
+              labelText={t('forms.berth.width')}
+              invalid={!!errors.width}
+              invalidText={errors.width}
+            />
+            <TextInput
+              id="length"
+              type="number"
+              onChange={handleChange}
+              value={String(values.length)}
+              labelText={t('forms.berth.length')}
+              invalid={!!errors.length}
+              invalidText={errors.length}
+            />
+            <TextInput
+              id="depth"
+              type="number"
+              onChange={handleChange}
+              value={String(values.depth)}
+              labelText={t('forms.berth.depth')}
+              invalid={!!errors.depth}
+              invalidText={errors.depth}
+              readOnly
+            />
+          </Grid>
+
+          <hr />
+
+          <Select
+            id="mooringType"
+            value={values.mooringType}
+            labelText={t('forms.berth.mooringType')}
+            options={Object.keys(BerthMooringType).map(mooringType => {
+              return {
+                label: t(`common.mooringTypes.${mooringType}`),
+                value: mooringType,
+              };
+            })}
+            onChange={handleChange}
+            required
+          />
+
+          <TextInput
+            id="comment"
+            onChange={handleChange}
+            value={values.comment}
+            labelText={t('forms.berth.comment')}
+          />
+          <Checkbox
+            id="isActive"
+            onChange={handleChange}
+            checked={values.isActive}
+            label={t('forms.berth.isActive')}
+          />
+          <div className={styles.formActionButtons}>
+            <Button
+              disabled={isSubmitting}
+              color={'supplementary'}
+              onClick={onCancel}
+            >
+              {t('forms.common.cancel')}
+            </Button>
+            <Button
```

Comment:
Forgot this button should be moved to the top left of the form as in the designs. Will fix.

Reply:
Done.